aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -140,6 +140,41 @@ bool is_nonzero(const Tensor& self) {
   TORCH_INTERNAL_ASSERT(false, "Expected non-Tensor backend scalar");
 }
 
+namespace {
+
+inline at::Tensor scalar_to_tensor_default_dtype(
+    Scalar s,
+    const Device device = at::kCPU) {
+  if (s.isFloatingPoint()) {
```

Comment:
Nit: These conditions should probably just set a dtype value so you can have a single `return at::scalar_tensor...` line at the end of the function.